ssh.go

```diff
@@ -324,48 +326,50 @@ func attachCommands(ssh *sshd.SSHServer, hostMap *HostMap, pendingHostMap *HostM
 func sshListHostMap(hostMap *HostMap, a interface{}, w sshd.StringWriter) error {
 	fs, ok := a.(*sshListHostMapFlags)
 	if !ok {
-		//TODO: error
-		return nil
+		return fmt.Errorf("flags not of type *sshListHostMapFlags")
+	}
+
+	type hostInfo struct {
+		VpnIP          net.IP                  `json:"vpnIp"`
+		LocalIndex     uint32                  `json:"localIndex"`
+		RemoteIndex    uint32                  `json:"remoteIndex"`
+		RemoteAddrs    []*udpAddr              `json:"remoteAddrs"`
+		CachedPackets  int                     `json:"cachedPackets"`
+		Cert           *cert.NebulaCertificate `json:"cert"`
+		MessageCounter *uint64                 `json:"messageCounter"`
 	}
 
 	hostMap.RLock()
-	defer hostMap.RUnlock()
+	hosts := make(map[uint32]hostInfo, len(hostMap.Hosts))
```

Comment:
I am unsure if we'd be willing to change this now, but we could turn this into an array and speed up/simplify this flow.